src/mantle/testing/concerns/trait-phpunit-upgrade-warning.php

```diff
@@ -0,0 +1,131 @@
+	 *
+	 * @var bool
+	 */
+	public bool $silence_phpunit_warning = false;
+
+	/**
+	 * Check if the current codebase is running PHPUnit 10 or higher.
+	 *
+	 * @return bool
+	 */
+	protected function is_running_phpunit_10_or_higher(): bool {
+		// Prevent conflicts if the internal class API changes.
+		if ( ! class_exists( Version::class ) || ! method_exists( Version::class, 'id' ) ) {
+			return false;
+		}
+
+		return version_compare( '10.0.0', Version::id(), '<=' );
+	}
+
+	/**
+	 * Warn the user if they're running PHPUnit 10+ against an incompatible test suite.
+	 *
+	 * @param TestSuite $test_suite Test suite configuration.
+	 */
+	protected function warn_if_test_suite_contains_legacy_test_cases( TestSuite $test_suite ): bool {
+		// Check if the test suite contains directories with a 'test-' prefix. That
+		// is a clear sign of a legacy test suite.
+		$legacy_test_cases = collect( $test_suite->directories() )
+			->filter(
+				fn ( $directory ) => method_exists( $directory, 'prefix' ) && 'test-' === $directory->prefix(),
+			);
+
+		if ( $legacy_test_cases->is_empty() ) {
+			return false;
+		}
+
+		render(
+			<<<HTML
+			<div class="space-y-1">
+				<div class="bg-red-300 text-red-700 pt-2 py-2">
+					<strong>🚨 Warning:</strong> You are running PHPUnit 10+ against a test suite that contains legacy test cases.
+				</div>
+				<div>
+					<span class="text-blue-300">Mantle Testing Framework 1.0</span> includes <span class="text-yellow-500 font-bold">✨ PHPUnit 10 ✨</span> which requires test cases to follow PSR-4 standards.
+					<br />
+					For example, that would be <span class="italic">tests/Feature/MyExampleTest.php</span> instead of <span class="italic">tests/feature/test-my-example.php</span>.
+					<br />
+					Fear not, you don't have to upgrade to PHPUnit 10 right away. You can still use PHPUnit 9 by running the following code:
+
+					<div class="ml-2 mt-1 italic">composer require --dev phpunit/phpunit:^9 nunomaduro/collision:^6</div>
+				</div>
+				<div>
+					For more information and tips on how to upgrade your codebase to PHPUnit 10, please refer to the 1.0 Release Changelog:
+
+					<div class="ml-2 my-1 italic">
+						https://github.com/alleyinteractive/mantle-framework/blob/1.x/CHANGELOG.md#phpunit-10-migration
+					</div>
+				</div>
+			</div>
+			HTML,
+		);
+
+		return true;
+	}
+
+	/**
+	 * Warn the user if they're running PHPUnit 10+ against an incompatible codebase.
+	 */
+	protected function warn_if_phpunit_10_or_higher(): void {
+		if ( $this->silence_phpunit_warning ) {
+			return;
+		}
+
+		if ( ! $this->is_running_phpunit_10_or_higher() ) {
+			return;
+		}
+
+		if ( ! class_exists( Registry::class ) || ! method_exists( Registry::class, 'get' ) ) {
+			return;
+		}
+
+		$registry = Registry::get();
+
+		foreach ( $registry->testSuite()->asArray() as $test_suite ) {
+			if ( $test_suite instanceof TestSuite && $this->warn_if_test_suite_contains_legacy_test_cases( $test_suite ) ) {
```

Comment:
`warn_if_test_suite_contains_legacy_test_cases` outputs a message and returns a boolean, is that a problem here? should those two concerns be separated into two functions?